src/core/thread/mle_router.cpp

```diff
@@ -1774,6 +1777,40 @@ void MleRouter::HandleStateUpdateTimer(void)
         break;
     }
 
+    if (mIsRouterRestoringChildren)
```

Comment:
`HandleStateUpdateTimer()` is called every second (`kStateUpdatePeriod   = 1000`). Each time in the loop below we send a "Child Update Request" to one child and wait for next timer callback for the next child. 
With the new code, if we have 32 children, it takes 32s to to send the "Child Update" to the last child (which I think is too long). 

I suggest to keep the "child update request timer" for this purpose and use a shorter interval.